plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1919,7 +1919,7 @@ public void finishMerge(ConnectorSession session, ConnectorMergeTableHandle merg
             }
 
             for (String file : oldFiles) {
-                transactionLogWriter.appendRemoveFileEntry(new RemoveFileEntry(file, writeTimestamp, true));
+                transactionLogWriter.appendRemoveFileEntry(new RemoveFileEntry(toUriFormat(file), writeTimestamp, true));
```

Comment:
There's another method `executeDelete` doing DELETE statement. Please add a new test case.

Reply:
I've extended the current test to also do a non-targeted `DELETE FROM`.